luarules/gadgets/unit_landspeedmultiplier.lua

```diff
@@ -0,0 +1,102 @@
+local gadget = gadget ---@type Gadget
+
+function gadget:GetInfo()
+    return {
+        name      = "Land Speed Multiplier",
+        desc      = "Speeds up or slows down units on land compared to their default water speed.",
+        author    = "ZephyrSkies, with a lot of help from [BONELESS]/qscrew/efrec",
+        date      = "2025-09-14",
+        license   = "GNU GPL, v2 or later",
+        layer     = 0,
+        enabled   = true
+    }
+end
+
+-- units need to have the following customparam set for this to work properly:
+-- landspeedfactor (number, e.g. 0.5 = half speed on land, 1 = no change, 2 = double speed)
```

Comment:
why not make it a water multiplier instead? land is the "base" domain, water is secondary